contracts/marketplace/src/execute.rs

```diff
@@ -431,42 +432,151 @@ pub fn execute_refund_renewal(
 
+    // make sure we're not trying to renew in the future
     if time > env.block.time {
         return Err(ContractError::CannotProcessFutureRenewal {});
     }
 
-    // // TODO: add renewal processing logic
-    // let renewal_queue = RENEWAL_QUEUE.load(deps.storage, time)?;
-    // for name in renewal_queue.iter() {
-    //     let ask = asks().load(deps.storage, ask_key(name))?;
-    //     if ask.renewal_fund.is_zero() {
-    //         continue;
-    //         // transfer ownership to name service
-    //         // list in marketplace for 0.5% of bid price
-    //         // if no bids, list for original price
-    //     }
-
-    //     // charge renewal fee
-    //     // pay out reward to operator
-    //     // reset ask
-
-    //     // Update Ask with new renewal_time
-    //     let renewal_time = env.block.time.plus_seconds(SECONDS_PER_YEAR);
-    //     let ask = Ask {
-    //         token_id: name.to_string(),
-    //         id: ask.id,
-    //         seller: ask.seller,
-    //         renewal_time,
-    //         renewal_fund: ask.renewal_fund - payment, // validate payment
-    //     };
-    //     store_ask(deps.storage, &ask)?;
-    // }
+    // get the queue to process
+    let renewal_queue = RENEWAL_QUEUE.load(deps.storage, (time.seconds(), 0))?;
+    // if renewal queue is empty we error out
+    if renewal_queue.is_empty() {
+        print!("Queue is empty");
+        return Err(ContractError::NoRenewalQueue {});
+    }
+    // fetch all items in the renewal queue
+    let items: Result<Vec<((u64, u64), String)>, cosmwasm_std::StdError> = RENEWAL_QUEUE
+        .range(
+            deps.as_ref().storage,
+            None,
+            None,
+            cosmwasm_std::Order::Ascending,
+        )
+        .collect();
+
+    // iterate through the items in the queue
+    // and process the renewal for each of them
+    for item in items.unwrap() {
+        // TODO: remove println debug statement
+        println!("Processing renewal queue item: {:?}", item.1);
+        let res = Response::new();
+        let name = item.1;
+
+        // fetch the asks and if there aren't any
+        // burn the name
+        let asks_for_name = asks()
+            .range(
+                deps.as_ref().storage,
+                Some(Bound::inclusive(ask_key(&name))),
+                None,
+                Order::Ascending,
+            )
+            .collect::<Vec<_>>();
+        if asks_for_name.is_empty() {
+            let burn_msg = Cw721ExecuteMsg::Burn {
+                token_id: name.to_string(),
+            };
+            let exec_burn_msg = WasmMsg::Execute {
+                contract_addr: collection.to_string(),
+                msg: to_binary(&burn_msg)?,
+                funds: vec![],
+            };
+
+            res.add_event(Event::new("burn").add_attribute("token_id", name.to_string()))
+                .add_message(exec_burn_msg);
+        } else {
```

Comment:
I guess this should never happen since there must always be an ask for a name right ?